### PR TITLE
Implement logic for the `max_latency_tier2` field

### PR DESCRIPTION
**Pull request by dpoltavets**
Closes: #4029

### CODE REVIEW

File: src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/PathFinder.java [hunk 1]
```diff
@@ -34,19 +33,19 @@
...
-                                                   SwitchId startSwitchId, SwitchId endSwitchId,
-                                                   WeightFunction weightFunction)
+    FindPathResult findPathInNetwork(AvailableNetwork network,
+                                     SwitchId startSwitchId, SwitchId endSwitchId,
+                                     WeightFunction weightFunction)
             throws UnroutableFlowException;
 
     /**
      * Finds a path whose weight is less than maxWeight and as close to maxWeight as possible.
      *
```
Comment (rozdy) [inline src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/PathFinder.java:43]: Add info about `backUpMaxWeight` to javadoc please.

File: src-java/kilda-pce/src/test/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinderTest.java [hunk 2]
```diff
@@ -261,43 +268,105 @@ public void shouldReturnTheShortestPath() throws  UnroutableFlowException {
...
+        assertFalse(pathResult.isBackUpPathComputationWayUsed());
+    }
+
+    /**
+     * System picks path closest to maxWeightTier2. Omit too cheap path and maxWeight path,
+     * omit equal to backUpMaxWeight path
+     */
+    @Test
+    public void shouldReturnThePathClosestToBackUpMaxWeight() throws UnroutableFlowException {
+        // a path with maxWeight 201
```
Comment (rozdy) [inline src-java/kilda-pce/src/test/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinderTest.java:283]: maxWeight -> backUpMaxWeight

**Review by rozdy — APPROVED** [review]

**Review by niksv — APPROVED** [review]

File: src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/PathFinder.java (hunk 1 above)
Comment (dpoltavets) [reply src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/PathFinder.java:44]: Added.

File: src-java/kilda-pce/src/test/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinderTest.java (hunk 2 above)
Comment (dpoltavets) [reply src-java/kilda-pce/src/test/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinderTest.java]: Fixed.

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/create/action/CompleteFlowCreateAction.java [hunk 3]
```diff
@@ -49,17 +49,42 @@ protected void perform(State from, State to, Event event, FlowCreateContext cont
...
+                }
+
+                flowPathRepository.updateStatus(stateMachine.getProtectedForwardPathId(), protectedPathStatus);
+                flowPathRepository.updateStatus(stateMachine.getProtectedReversePathId(), protectedPathStatus);
             }
-            flowRepository.updateStatus(flowId, FlowStatus.UP);
+
+            flowRepository.updateStatus(flowId, status);
+            if (FlowStatus.DEGRADED.equals(status)) {
+                flowRepository.updateStatusInfo(flowId, "A backup way was used to build the path");
```
[the remark this answers is not in the document]
Comment (dpoltavets) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/create/action/CompleteFlowCreateAction.java]: Status info message was updated.

**Pull request merged**